promql/engine.go

```diff
@@ -218,22 +218,27 @@ func contextDone(ctx context.Context, env string) error {
 // Engine handles the lifetime of queries from beginning to end.
 // It is connected to a querier.
 type Engine struct {
-	// The querier on which the engine operates.
-	querier local.Querier
+	// A Querier constructor against an underlying storage.
+	queryable Queryable
```

Comment:
I mean, I guess I know why, because then you can pass in the whole storage as a `Queryable`, but you could as well just pass in `storage.NewQuerier`.